conjure-core/src/main/java/com/palantir/conjure/defs/validator/ErrorDefinitionValidator.java

```diff
@@ -33,5 +36,15 @@ public static void validate(ErrorDefinition definition) {
                 Stream.concat(definition.getSafeArgs().stream(), definition.getUnsafeArgs().stream())
                         .map(FieldDefinition::getFieldName)
                         .collect(Collectors.toSet()));
+        ImmutableList<FieldName> fieldsDeclaringSafety = Stream.concat(
+                        definition.getSafeArgs().stream(), definition.getUnsafeArgs().stream())
+                .filter(arg -> arg.getSafety().isPresent())
+                .map(FieldDefinition::getFieldName)
+                .collect(ImmutableList.toImmutableList());
+        if (!fieldsDeclaringSafety.isEmpty()) {
+            throw new ConjureIllegalStateException("ErrorDefinition field safety is defined by the key 'safeArgs' or "
```

Comment:
Not sure I get the purpose of this validation. In `ErrorDefinition`'s, we don't allow specifying LogSafety because we already have `safeArgs` and `unsafeArgs` as a first-class concept?

Is it a use-case that people reuse `FieldDefinition` between `ErrorDefinition` and other parts of their API? Afaik, this would limit such reuse? Maybe we want to validate that definitions used as `safeArg` only have safe annotations?

Reply:
FieldDefinition from the IR is reused several places within the IR (but does not allow definitions themselves to be used externally). These include : errordefinition safe/unsafe args, object fields, and union types.

I considered breaking out a new type for errors which doesn’t allow safety info, but figured the Abi change wasn’t worth it because we also need to validate that safety isn’t used with complex objects, similar validation for error defs is simple enough. Not a strong opinion, what do you think?

Reply:
Thanks for explaining, makes sense. Yeah, think its fine to leave it as-is with validation instead of introducing the new type if this would be a larger change!